src/crawlee/_utils/urls.py

```diff
@@ -1,25 +1,31 @@
 from __future__ import annotations
 
-from urllib.parse import parse_qs, urljoin, urlparse
+from typing import TYPE_CHECKING
 
 from pydantic import AnyHttpUrl, TypeAdapter
+from yarl import URL
+
+if TYPE_CHECKING:
+    from multidict import MultiDictProxy
 
 
 def is_url_absolute(url: str) -> bool:
     """Check if a URL is absolute."""
-    url_parsed = urlparse(url)
-    return bool(url_parsed.scheme) and bool(url_parsed.netloc)
+    url_parsed = URL(url)
+
+    # We don't use .absolute because in yarl.URL, it is always True for links that start with '//'
+    return bool(url_parsed.scheme) and bool(url_parsed.raw_authority)
 
 
 def convert_to_absolute_url(base_url: str, relative_url: str) -> str:
     """Convert a relative URL to an absolute URL using a base URL."""
-    return urljoin(base_url, relative_url)
+    return str(URL(base_url).join(URL(relative_url)))
 
 
-def extract_query_params(url: str) -> dict[str, list[str]]:
+def extract_query_params(url: str) -> MultiDictProxy[str]:
```

Comment:
I'd argue this function could be removed - `extract_query_params(url)` does not seem to have an advantage over `URL(url).query` IMO.